client/my-sites/checkout/src/components/checkout-main.tsx

```diff
@@ -10,13 +9,6 @@ import debugFactory from 'debug';
 import { useTranslate } from 'i18n-calypso';
 import { Fragment, useCallback, useMemo } from 'react';
 import QueryContactDetailsCache from 'calypso/components/data/query-contact-details-cache';
```

Comment:
Is there a reason that we're keeping this one? It seems to just check if there are any contact details cached and I suppose provides them as a local object to the checkout steps? 

But it also looks like we query the contact details elsewhere in the file anyway https://github.com/Automattic/wp-calypso/blob/fc02c411ef7d41bbecbf6d59c4da13a57767e016/client/my-sites/checkout/src/components/checkout-main.tsx#L391

I know contact details are a relatively complex issue in the cart since they can be provided by the domain and also provided by the customer at checkout 🤔

Reply:
Hm... `useSelect` refers to the checkout store, which is a `@wordpress/data` store used for checkout's current state (eg: what the user has typed into their contact details fields). The "cached contact details" data queried by `QueryContactDetailsCache` is the _saved_ contact details from a previous checkout experience, fetched from the API and saved into Redux. The `useCachedDomainContactDetails()` hook copies the data from the Redux store into the `@wordpress/data` store, so we still need to fetch it for Redux.

However! I just noticed that `useCachedDomainContactDetails()` can actually trigger the fetch itself, which means that we probably don't need this one either. I'll remove it.